### PR TITLE
Feature/autonumber relationships

### PR DESCRIPTION
**Pull request by daniel-simpson**
Implements #115

### CODE REVIEW

File: LayoutOptions.md [hunk 1]
```diff
@@ -139,4 +139,28 @@ Rel(web_app, twitter, "Gets tweets from", "HTTPS")
...
+}
+System(twitter, "Twitter")
+
+Rel(admin, web_app, "Uses", "HTTPS")
+Rel(web_app, twitter, "Gets tweets from", "HTTPS")
+Rel_Right(web_app, db, "Stores", "SQL")
+@enduml
+```
+
+![auto number relationships example](images/AUTONUMBER_RELATIONSHIPS_Sample.png)
```
Comment (daniel-simpson) [inline LayoutOptions.md:166]: Note: I'm unsure how to generate the `https://www.plantuml.com/plantuml/png/IMAGEBINARY` style image preview here, so I've added a image itself to the PR.

Happy to remove the last commit and replace with the same URL as above if someone can help me with generating one.

Thanks!

Comment (adrianvlupu) [reply LayoutOptions.md:166]: In order to get a image and include it in a markdown file, you can 
1 go to http://www.plantuml.com/plantuml/ 
2 paste your diagram code
3 Submit 
4 View as png
5 add `diagram` to the markdown file

**Comment by adrianvlupu** [comment]
Isn't this already implemented in C4_Dynamic.puml? I'm curious what are the advantages of having a top level method versus the dynamic diagram

```plantuml
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Dynamic.puml

ContainerDb(c4, "Database", "Relational Database Schema", "Stores user registration information, hashed authentication credentials, access logs, etc.")
Container(c1, "Single-Page Application", "JavaScript and Angular", "Provides all of the Internet banking functionality to customers via their web browser.")
Container_Boundary(b, "API Application") {
  Component(c3, "Security Component", "Spring Bean", "Provides functionality Related to signing in, changing passwords, etc.")
  Component(c2, "Sign In Controller", "Spring MVC Rest Controller", "Allows users to sign in to the Internet Banking System.")
}
Rel_R(c1, c2, "Submits credentials to", "JSON/HTTPS")
Rel(c2, c3, "Calls isAuthenticated() on")
Rel_R(c3, c4, "select * from users where username = ?", "JDBC")
@enduml
```

C4_Dynamic Diagram Sample - bigbankplc

**Comment by daniel-simpson** [comment]
To be honest, I am still pretty new to C4 and PlantUML and hadn't looked in C4_Dynamic or C4_Deployment yet.

This definitely solves my use case, happy for this issue to be closed.

Thanks!

**Comment by daniel-simpson** [comment]
I did find one potential advantage to having this as a top-level method rather than only available in C4_Dynamic (though its only pretty minor): Legend complexity.

If I'm creating a Context Diagram and want both number relationships and show a legend, I see erroneous types in the legend (container, component, external container, external component) as C4_Dynamic imports all of these types regardless of whether i need them:

```plantuml
@startuml b2clogin_context
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Dynamic.puml

LAYOUT_WITH_LEGEND()

Person(user, "User", "A property manager, tenant, owner or supplier that needs access to a stable feature set")

System(software, "Software", "Provides new features regularly")

System_Boundary(azure, "Azure"){
  System_Ext(b2c, "Azure AD B2C", "Azure Active Directory Business-to-Consumer is Azure's platform offering encompassing scalable authentication and authorization for customer-facing applications")
  System_Ext(aad, "Azure AD", "Azure Active Directory is Azure's cloud-native user, group and permission storage offering")
}

Rel(user, software, "Uses")
Rel_Right(software, b2c, "Redirects user to")
Rel(user, b2c, "Logs in using")
Rel(b2c, aad, "Retrieves user information from")
Rel_Left(b2c, software, "Redirects with JWT back to")
@enduml
```

example diagram with extra legend elements

**Comment by Potherca** [comment]
Hmmm... That could potentially be a bug :thinking:

**Comment by adrianvlupu** [comment]
Not really a bug, c4_dynamic.puml includes c4_component.puml, but this could be worked out

**Comment by kirchsth** [comment]
The (dynamic) legend problem is solved already with the new integrated SHOW_DYNAMIC_LEGEND().
(Please don't forget it has to be the last line)

@adrianvlupu: I think it can be closed again
@daniel-simpson: If you read the new README.md and LayoutOptions.md you can find more details

```
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Dynamic.puml

Person(user, "User", "A property manager, tenant, owner or supplier that needs access to a stable feature set")

System(software, "Software", "Provides new features regularly")

System_Boundary(azure, "Azure"){
  System_Ext(b2c, "Azure AD B2C", "Azure Active Directory Business-to-Consumer is Azure's platform offering encompassing scalable authentication and authorization for customer-facing applications")
  System_Ext(aad, "Azure AD", "Azure Active Directory is Azure's cloud-native user, group and permission storage offering")
}

Rel(user, software, "Uses")
Rel_Right(software, b2c, "Redirects user to")
Rel(user, b2c, "Logs in using")
Rel(b2c, aad, "Retrieves user information from")
Rel_Left(b2c, software, "Redirects with JWT back to")

SHOW_DYNAMIC_LEGEND()
@enduml
```
with dynamic legen

**Pull request closed without merging**